src/App.tsx

```diff
@@ -1,53 +1,86 @@
-import React from 'react';
+import React, { useMemo, useState } from 'react';
 import './App.scss';
+import cn from 'classnames';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+
+const debounce = (f:(...args: string[]) => void,
+  delay: number) => {
+  let timerId: NodeJS.Timeout;
+
+  return (...arg: string[]) => {
+    clearTimeout(timerId);
+    timerId = setTimeout(f, delay, ...arg);
+  };
+};
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  const applyQuery = debounce(setAppliedQuery, 1000);
+
+  const clearHandler = () => {
+    setQuery('');
+    setAppliedQuery('');
+  };
+
+  const personFilter = useMemo(() => {
+    return peopleFromServer.filter(
+      people => people.name.toLowerCase().includes(appliedQuery.toLowerCase()),
+    );
+  }, [appliedQuery]);
 
   return (
     <main className="section">
       <h1 className="title">
-        {`${name} (${born} = ${died})`}
+        {selectedPerson
+          ? (`${selectedPerson.name} (${selectedPerson.born} - ${selectedPerson.died})`)
+          : ('No person is selected')}
       </h1>
 
-      <div className="dropdown is-active">
+      <div className={cn('dropdown', {
+        'is-active': appliedQuery,
+      })}
+      >
         <div className="dropdown-trigger">
           <input
             type="text"
             placeholder="Enter a part of the name"
             className="input"
+            value={query}
+            onChange={event => {
+              setQuery(event.target.value);
+              applyQuery(event.target.value);
+            }}
           />
         </div>
 
         <div className="dropdown-menu" role="menu">
           <div className="dropdown-content">
             <div className="dropdown-item">
-              <p className="has-text-link">Pieter Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Bernard Haverbeke</p>
-            </div>
+              {personFilter.map((person) => (
+                <button
+                  className="button"
+                  type="button"
+                  key={person.slug}
+                  onClick={() => {
+                    setSelectedPerson(person); clearHandler();
```

Comment:
```suggestion
                    setSelectedPerson(person); 
                    clearHandler();
```
at least, I'd write each one in the new line